test/onnx/test_pytorch_onnx_shape_inference.py

```diff
@@ -283,5 +285,112 @@ def test_reduce_prod_without_axes(self):
         self.run_test(g, reduce_prod.node(), expect_tensor("Long", shape=(1,)))
 
 
+class TestONNXCustomOpShapeInference(common_utils.TestCase):
+
+    opset_version = _constants.ONNX_MAX_OPSET
+
+    def test_single_customOp_setType(self):
+
+        self.addCleanup(torch.onnx.unregister_custom_op_symbolic, "::linalg_inv", 9)
+
+        class CustomInverse(torch.nn.Module):
+            def forward(self, x):
+                return torch.inverse(x) + x
+
+        def linalg_inv_setType(g, self):
+            return g.op("com.microsoft::Inverse", self).setType(self.type())
+
+        torch.onnx.register_custom_op_symbolic("::linalg_inv", linalg_inv_setType, 9)
+        model = CustomInverse()
+        x = torch.randn(2, 3, 3)
+        f = io.BytesIO()
+        torch.onnx.export(
+            model,
+            (x,),
+            f,
+            opset_version=self.opset_version,
+            custom_opsets={"com.microsoft": 1},
+        )
+
+        model_proto = onnx.load(io.BytesIO(f.getvalue()))
+        dims = model_proto.graph.value_info[0].type.tensor_type.shape.dim
+        for i in range(len(dims)):
+            self.assertTrue(dims[i].HasField("dim_value"))
+        for dim, rank in zip(dims, x.size()):
+            self.assertEqual(dim.dim_value, rank)
+
+    def test_single_customOp_NO_setType(self):
```

Comment:
snake case, here and elsewhere when appropriate

Reply:
Done